qiskit/visualization/text.py

```diff
@@ -513,7 +513,10 @@ def lines(self, line_length=None):
         layers = self.build_layers()
 
         if not line_length:
-            line_length = self.line_length
+            if self.line_length:
+                line_length = self.line_length
+            else:
+                line_length, _ = get_terminal_size()
```

Comment:
It seems that there is some overlap with https://github.com/Qiskit/qiskit-terra/pull/2253/files#diff-1fd0d64450f10e7e32835e247a934235R503

Maybe line_length setting should happen in a single place...

Reply:
Sure, that works fine for me, I figured there was a reason for staging the checks like this which is why I just added another check to avoid the edge case. But if you want to just simplify the logic that works fine too,

Reply:
I simplified and combined the logic in: https://github.com/Qiskit/qiskit-terra/pull/2253/commits/2e18182a99e9ea32e9aa2d6cd7052dc644985d04